SquAPI.lua

```diff
@@ -73,1100 +68,1238 @@ squapi.autoFunctionUpdates = true
-    self.isBlink = isBlink
-    self.animation = animation
-	self.chanceRange = chanceRange or 200
-
-
-    -- CONTROL -------------------------------------------------------------------------
-	
-    self.enabled = true
-    function self:toggle()
-		self.enabled = not self.enabled
-	end
-    function self:disable()
-        self.enabled = false
-    end
-    function self:enable()
-        self.enabled = true
-    end
+  ---@class SquAPI.Randimation
+  local _self = setmetatable({}, squapi.randimation)
 
-    -- UPDATES -------------------------------------------------------------------------
+  -- INIT -------------------------------------------------------------------------
+  _self.isBlink = isBlink
+  _self.animation = animation
+  _self.chanceRange = chanceRange or 200
 
-	function events.tick()
-		if self.enabled and (not self.isBlink or player:getPose() ~= "SLEEPING") and math.random(0, self.chanceRange) == 0 and self.animation:isStopped() then
-            self.animation:play()
-		end
-	end
 
-	return self
-end
+  -- CONTROL -------------------------------------------------------------------------
+
+  _self.enabled = true
+  ---Toggle this randimation on or off
+  function _self:toggle()
+    _self.enabled = not _self.enabled
+  end
+
+  ---Disable this randimation
+  function _self:disable()
+    _self.enabled = false
+  end
+
+  ---Enable this randimation
+  function _self:enable()
+    _self.enabled = true
+  end
 
+  ---Sets if this randimation is enabled
+  ---@param bool boolean
+  function _self:setEnabled(bool)
+    assert(type(bool) == "boolean",
+      "§4setEnabled must be set to a boolean.§c")
+    _self.enabled = bool
+  end
 
--- MOVING EYES
---guide:(note if it has a * that means you can leave it blank to use reccomended settings)
--- element:	 		the eye element that is going to be moved, each eye is seperate.
--- *leftdistance: 	the distance from the eye to it's leftmost posistion
--- *rightdistance: 	the distance from the eye to it's rightmost posistion
--- *updistance: 	the distance from the eye to it's upmost posistion
--- *downdistance: 	the distance from the eye to it's downmost posistion
+  -- UPDATES -------------------------------------------------------------------------
+
+  ---Run tick function on randimation
+  function events.tick()
+    if _self.enabled and (not _self.isBlink or player:getPose() ~= "SLEEPING") and math.random(0, _self.chanceRange) == 0 and _self.animation:isStopped() then
+      _self.animation:play()
+    end
+  end
+
+  table.insert(squapi.randimations, _self)
+  return _self
+end
+
+---Contains all registered eyes
+---@type SquAPI.Eye[]
 squapi.eyes = {}
 squapi.eye = {}
 squapi.eye.__index = squapi.eye
+
+---MOVING EYES - Moves an eye based on the head rotation to look toward where you look; should work with any general eye type.<br><br>Note: you call this function for each eye, so if you have two eyes you will call this function twice (one for each eye).
```

Comment:
These two `<br>` tags should be replaced with a markdown paragraph separator
```suggestion
---MOVING EYES - Moves an eye based on the head rotation to look toward where you look; should work with any general eye type.
---
---Note: you call this function for each eye, so if you have two eyes you will call this function twice (one for each eye).
```